### PR TITLE
doc: Fix grammar and punctuation in developer notes

### PR DESCRIPTION
**Pull request by Tech1k**
This pull request is regarding minor grammar and punctuation errors in the developer notes. There were no modifications to the existing code, only alterations to fix the grammar and punctuation in the text to make the developer notes more understandable and easier to read.

### CODE REVIEW

File: doc/developer-notes.md [hunk 1]
```diff
@@ -424,7 +426,7 @@ to do this is thus to create your local gitignore. Add this to `~/.gitconfig`:
 (alternatively, type the command `git config --global core.excludesfile ~/.gitignore_global`
 on a terminal)
 
-Then put your favourite tool's temporary filenames in that file, e.g.
+Then put your favorite tool's temporary filenames in that file, e.g.
```
Comment (fanquake) [inline doc/developer-notes.md]: This can be spelt either way, so you can leave it as is.

File: doc/developer-notes.md [hunk 2]
```diff
@@ -633,22 +635,21 @@ AddressBookPage::AddressBookPage(Mode _mode)
...
+  configuring with `--enable-debug`.
 
 - When using `LOCK`/`TRY_LOCK` be aware that the lock exists in the context of
   the current scope, so surround the statement and the code that needs the lock
-  with braces
+  with braces.
 
   OK:
-
+
```
Comment (fanquake) [inline doc/developer-notes.md]: You've introduced white space here and below.

**Review by fanquake — COMMENTED** [review]
These changes look ok, however you'll need to fix up the trailing white space to appease the linter.

**Comment by fanquake** [comment]
Please squash your changes into a single commit. You commit message and description in the first commit is fine.

**Comment by Tech1k** [comment]
> Please squash your changes into a single commit. You commit message and description in the first commit is fine.

@fanquake Could I please have some assistance with squashing these commits before I mess everything up. I already have the pull request open for edits from maintainers. Thank you in advance.

**Comment by practicalswift** [comment]
@Tech1k `git rebase -i HEAD~4` then change from `pick` to `fixup` on L2-4. Exit editor and `git push -f`. Done! :-)

**Comment by Tech1k** [comment]
@practicalswift Did I get it just right or did I wreak more havoc to it?

**Comment by practicalswift** [comment]
@Tech1k The commits tab shows six commits -- it should be one when you've successfully squashed. Please see if https://github.com/wprig/wprig/wiki/How-to-squash-commits can help you out :-)

**Comment by Tech1k** [comment]
@practicalswift Thank you for pointing me in the right direction. I'm only 14 years old and I  need to 'git' my act together. Especially as I begin to write some unit tests for Bitcoin in the dialect of Python.

**Comment by Tech1k** [comment]
Whoops, I accidentally committed four other files that are not mine, let me revise that.

**Review by fanquake — APPROVED** [review]
ACK b2ea20d3302275a62bbdfdb96169c6788fe7b9c1

If you're looking for more ways to contribute, check out the [good first issue tag](https://github.com/bitcoin/bitcoin/issues?q=is%3Aopen+is%3Aissue+label%3A%22good+first+issue%22).

In future, when squashing commits, make sure you prune all of the intermediate commit messages from your commit body. i.e there are still multiple "Update developer-notes.md" lines left here.

**Pull request merged**